thrift/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactories.java

```diff
@@ -70,14 +70,28 @@ public String toString() {
     /**
      * {@link TProtocolFactory} for the Thrift TText protocol.
      */
-    public static final TProtocolFactory TEXT = TTextProtocolFactory.get();
+    public static final TProtocolFactory TEXT = TTextProtocolFactory.get(false);
+
+    /**
+     * {@link TProtocolFactory} for the Thrift TText protocol that serializes enums as strings.
+     */
+    public static final TProtocolFactory TEXT_ENUM = TTextProtocolFactory.get(true);
 
     /**
      * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
      *
      * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not for Thrift
      */
     public static TProtocolFactory get(SerializationFormat serializationFormat) {
+        return get(serializationFormat, false);
+    }
+
+    /**
+     * Returns the {@link TProtocolFactory} for the specified {@link SerializationFormat}.
+     *
+     * @throws IllegalArgumentException if the specified {@link SerializationFormat} is not for Thrift
+     */
+    public static TProtocolFactory get(SerializationFormat serializationFormat, boolean writeEnumsAsString) {
```

Comment:
This doesn't sound right to me because `writeEnumsAsString` is only useful for TEXT protocol. Howa bout reverting this and let a user refer to `TEXT_ENUM` directly?

Reply:
Sure! The reason I wrote this is that if you set `writeEnumsAsString` in `ThriftJacksonModule`, you might assume that behavior will work for all protocols

Reply:
I think it's fine since `TEXT_ENUM` is not really a default behavior. For less confusion, how about removing `TEXT_ENUM` and letting a user get the `TProtocolFactory` via `TTextProtocolFactory.get(true)`?

Reply:
Alternatively, we could register a new `SerializationFormat`? Then this problem could go away completely.

Reply:
e.g. `ttext_named_enum` and `TTEXT_NAMED_ENUM`

Reply:
Done. I had to name the protocol `ttext-named-enum` since underscores are not allowed.